### DISCUSSION TITLE
How to add a searchable ui.Input() element following a header text to filter Table data?

### DISCUSSION THREAD

**Discussion by happybeginning1** [zauberzeug/nicegui#3215] [Q&A] [answered]
### Question

Hello,

The following example code works well:

```python
from nicegui import ui

columns = [
    {'name': 'long_temp', 'label': 'long_temp', 'field': 'long_temp', 'align': 'center'},
    {'name': 'type', 'label': 'type', 'field': 'type', 'align': 'center'},
    {'name': 'time', 'label': 'time', 'field': 'time', 'align': 'center'}
]
    
data_list = [
    {'long_temp': 'long', 'type': 'dinosaur', 'time': '08-05 17:31'},
    {'long_temp': 'long', 'type': 'cat', 'time': '11-01 13:20'},
    {'long_temp': 'long', 'type': 'dog', 'time': '08-25 00:37'},
    {'long_temp': 'temp', 'type': 'sheep', 'time': '04-19 17:26'},
    {'long_temp': 'temp', 'type': 'fish', 'time': '03-10 11:28'}
]

ui.table(columns=columns, rows=data_list, row_key='time').classes('w-full')

ui.run()
```

I want to add an ui.input() element to the right of the header text "type" (with a cetain gap between the two elements, as shown in the example image below). So that when the user enters data into this text box, the table will filter and display only the items that contain the entered keyword.

I have checked some documentation and discussions, but I couldn't find a workable solution. Could you please guide me on how to achieve this functionality?

Thank you very much!

![example](https://github.com/zauberzeug/nicegui/assets/168985300/f502f871-5092-43da-9452-2a8618daacde)

**Comment by falkoschindler** [comment, marked as answer]
Hi @happybeginning1,

You can overwrite the header cell slot for the "type" column, insert an input element and emit a custom event when the enter key is pressed:
```py
table = ui.table(columns=columns, rows=data_list, row_key='time').classes('w-full')
table.add_slot('header-cell-type', '''
    <th class="flex flex-row items-center gap-4 justify-center">
        type
        <q-input
            v-model="filterValue"
            placeholder="filter"
            @keydown.enter="$parent.$emit('foo', filterValue)"
        />
    </th>
''')
table.on('foo', lambda e: print('Filter:', e.args))
```
![Screenshot 2024-06-13 at 08 07 58](https://github.com/zauberzeug/nicegui/assets/5767091/f521cb72-9dd0-474b-851a-aa3e8541795c)

**Reply by happybeginning1** [reply]
Hi @falkoschindler,
It works. Thank you for the quick response!
